app-crypt/libsecret/libsecret-0.21.2.ebuild

```diff
@@ -0,0 +1,164 @@
+
+DEPEND="
+	>=dev-libs/glib-2.44:2[${MULTILIB_USEDEP}]
+	crypt? ( >=dev-libs/libgcrypt-1.2.2:0=[${MULTILIB_USEDEP}] )
+	gnutls? ( >=net-libs/gnutls-3.8.2:0=[${MULTILIB_USEDEP}] )
+	tpm? ( >=app-crypt/tpm2-tss-3.0.3:= )
+	introspection? ( >=dev-libs/gobject-introspection-1.54:= )
+"
+RDEPEND="${DEPEND}"
+PDEPEND="virtual/secret-service"
+BDEPEND="
+	app-text/docbook-xml-dtd:4.2
+	dev-libs/libxslt
+	dev-util/gdbus-codegen
+	dev-util/glib-utils
+	>=sys-devel/gettext-0.19.8
+	virtual/pkgconfig
+	gtk-doc? (
+		app-text/docbook-xml-dtd:4.1.2
+		>=dev-util/gi-docgen-2021.7
+	)
+	test? (
+		$(python_gen_any_dep '
+			dev-python/dbus-python[${PYTHON_USEDEP}]
+			introspection? ( dev-python/pygobject:3[${PYTHON_USEDEP}] )')
+		test-rust? ( introspection? ( >=dev-libs/gjs-1.32 ) )
+		tpm? (
+			app-crypt/swtpm
+			app-crypt/tpm2-abrmd
+			>=app-crypt/tpm2-tss-3.2.0:=
+		)
+	)
+	vala? ( $(vala_depend) )
+"
+
+dbus_run() {
+	(
+		# start isolated dbus session bus
+		dbus_data=$(dbus-launch --sh-syntax) || exit
+		eval "${dbus_data}"
+
+		$@
+		ret=${?}
+
+		kill "${DBUS_SESSION_BUS_PID}"
+		exit "${ret}"
+	) || die
+}
+
+tpm2_run_with_emulator() {
+	export XDG_CONFIG_HOME=${T}/.config/swtpm
+	"${BROOT}"/usr/share/swtpm/swtpm-create-user-config-files --overwrite || die
+
+	mkdir -p ${XDG_CONFIG_HOME}/mytpm1 || die
+	swtpm_setup_args=(
+		--tpm2
+		--tpmstate ${XDG_CONFIG_HOME}/mytpm1
+		--createek
+		--allow-signing
+		--decryption
+		--create-ek-cert
+		--create-platform-cert
+		--lock-nvram
+		--overwrite
+		--display
+	)
+	swtpm_setup "${swtpm_setup_args[@]}" || die
+
+	swtpm_socket_args=(
+		--tpm2
+		--tpmstate dir=${XDG_CONFIG_HOME}/mytpm1
+		--flags startup-clear
+		--ctrl type=unixio,path=${XDG_CONFIG_HOME}/mytpm1/swtpm.socket.ctrl
+		--server type=unixio,path=${XDG_CONFIG_HOME}/mytpm1/swtpm.socket
+		--pid file=${XDG_CONFIG_HOME}/mytpm1/swtpm.pid
+		--daemon
+	)
+	swtpm socket "${swtpm_socket_args[@]}" || die
+
+	tpm2_abrmd_args=(
+		--logger=stdout
+		--tcti=swtpm:path=${XDG_CONFIG_HOME}/mytpm1/swtpm.socket
+		--session
+		--flush-all
+	)
+	tpm2-abrmd "${tpm2_abrmd_args[@]}" &
+	export TCTI=tabrmd:bus_type=session
+
+	$@ || die
+
+	# When swtpm dies, tmp2-abrmd will exit
+	kill $(< ${XDG_CONFIG_HOME}/mytpm1/swtpm.pid) || die
+}
+
+python_check_deps() {
+	if use introspection; then
+		python_has_version "dev-python/pygobject:3[${PYTHON_USEDEP}]" || return
+	fi
+	python_has_version "dev-python/dbus-python[${PYTHON_USEDEP}]"
+}
+
+pkg_setup() {
+	use test && python-any-r1_pkg_setup
+}
+
+src_prepare() {
+	use vala && vala_setup
+	default
+}
+
+multilib_src_configure() {
+	local emesonargs=(
+		$(meson_native_true manpage)
+		$(meson_native_use_bool vala vapi)
+		$(meson_native_use_bool gtk-doc gtk_doc)
+		$(meson_native_use_bool introspection)
+		-Dbashcompdir="$(get_bashcompdir)"
+		$(meson_native_enabled bash_completion)
+		$(meson_native_use_bool tpm tpm2)
+	)
+
+	if use crypt ; then
+        emesonargs+=( -Dcrypto=libgcrypt )
```

Comment:
I don't think crypt is a good USE flag name for this, because with `USE=-crypt`, you might expect the functionality is entirely disabled, but it isn't - you need `USE="-crypt -gnutls` for that.

Does it make sense for us to even make this optional? Do people want "transport encryption" off? (I'm not sure what it does in the context of libsecret).

If it makes sense for it to be optional, we should do the following to match how we handle imagemagick/graphicsmagick, and so on for "provider" USE flags:
* USE="crypt" (maybe? better names welcome...) => libgcrypt
* USE="crypt gnutls" => gnutls
* USE="gnutls" => ignored

If it doesn't make sense for it to be optional, we should do:
* USE=gnutls => use gnutls
* USE=-gnutls => use libgcrypt

Reply:
I agree, gcrypt can be a better alternative.
USE=+crypt was already present
There are already several packages depending on app-crypt/libsecret[crypt] , should we rename the use flag on every one?

Maybe transport encryption off make sense in some testing scenarios, since in upstream is optional maybe we should keep it optional with +crypt default

Reply:
Looking to the meson.build diff and this:
+option('crypto', type: 'combo', choices: ['libgcrypt', 'gnutls', 'disabled'], value: 'libgcrypt', description: 'Backend library to implement transport encryption')


I would opt for:
USE=gcrypt -> enables libgcrypt
USE=gnutls -> enables gnutls
Both USEs disable, pass "disabled" to crypto option

For the defaults, I would opt for enabling gcrypt (following upstream defaults)

Thanks

Reply:
> Looking to the meson.build diff and this: +option('crypto', type: 'combo', choices: ['libgcrypt', 'gnutls', 'disabled'], value: 'libgcrypt', description: 'Backend library to implement transport encryption')
> 
> I would opt for: USE=gcrypt -> enables libgcrypt USE=gnutls -> enables gnutls Both USEs disable, pass "disabled" to crypto option
> 
> For the defaults, I would opt for enabling gcrypt (following upstream defaults)
> 
> Thanks

If we change crypt by gcrypt? What will happen to all packages depending on app-crypt/libsecret-0.5[crypt] ?

Reply:
Good catch.

Are all those packages compatible with gnutls too? It seems they are from what I have seen. If that is the case, probably we should opt for the first Sam's suggestion:
"If it makes sense for it to be optional, we should do the following to match how we handle imagemagick/graphicsmagick, and so on for "provider" USE flags:

USE="crypt" (maybe? better names welcome...) => libgcrypt
USE="crypt gnutls" => gnutls
USE="gnutls" => ignored"